core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -718,6 +719,7 @@ private[spark] object JsonProtocol {
     metrics.setHostname((json \ "Host Name").extract[String])
     metrics.setExecutorDeserializeTime((json \ "Executor Deserialize Time").extract[Long])
     metrics.setExecutorRunTime((json \ "Executor Run Time").extract[Long])
+    metrics.setExecutorCpuTime((json \ "Executor CPU Time").extract[Long])
```

Comment:
what happens when the history server is tries to deserialise a history from an earlier spark version, one which doesn't have a CPU time? As it looks to me through my scan through the code that this is going to fail.

HistoryServerSuite is the regression test here —it does have job histories without the relevant metric.

It would benefit from having another reference test run here for playback